packages/iov-bcp-types/src/connection.ts

```diff
@@ -124,10 +139,40 @@ export interface ConfirmedTransaction<T extends UnsignedTransaction = UnsignedTr
   readonly transactionId: TransactionId;
   /** application specific data from executing tx (result, code, tags...) */
   readonly result?: Uint8Array;
+  /**
+   * Application specific logging output in an arbitrary text format that
+   * may change at any time.
+   */
   readonly log?: string;
   // readonly tags: ReadonlyArray<Tag>;
 }
 
+export interface FailedTransaction {
```

Comment:
Right. There more common fields `ConfirmedTransaction | FailedTransaction`, the easier it is to work on the union type instead of handling case-by-case.